### PR TITLE
quicklispPackages: update, sbcl: 2.0.8 -> 2.1.2, and related stuff

### PR DESCRIPTION
**Pull request by 7c6f434c**
###### Motivation for this change

SBCL was frozen to an older version because of unclear status of some packages with 2.0.9, and everything was stale.

Now everything that has ever worked in Nixpkgs with SBCL works with the latest. Not yet dropping the older versions, but won't object to doing it now.

Cleaned up a bit the situation for other Common Lisp implementations. CCL now builds everything but StumpWM; `iterate` works with all Common Lisp implementations we have. Embarassingly, it was not true before. Improved internal tooling to handle implementation-specific dependencies and fixes.

Will self-merge after tests pass.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/lisp-modules/quicklisp-to-nix-output/jonathan.nix [hunk 1]
```diff
@@ -1,15 +1,15 @@
...
-  description = "High performance JSON encoder and decoder. Currently support: SBCL, CCL.";
+  description = ''High performance JSON encoder and decoder. Currently support: SBCL, CCL.'';
 
   deps = [ args."alexandria" args."babel" args."cffi" args."cffi-grovel" args."cffi-toolchain" args."cl-annot" args."cl-ppcre" args."cl-syntax" args."cl-syntax-annot" args."fast-io" args."named-readtables" args."proc-parse" args."static-vectors" args."trivial-features" args."trivial-gray-streams" args."trivial-types" ];
 
   src = fetchurl {
-    url = "http://beta.quicklisp.org/archive/jonathan/2020-09-25/jonathan-20200925-git.tgz";
-    sha256 = "1y5v3g351nsy7px0frdr2asmcy0lyfbj73ic1f5yf4q65hrgvryx";
+    url = ''http://beta.quicklisp.org/archive/jonathan/2020-09-25/jonathan-20200925-git.tgz'';
+    sha256 = ''1y5v3g351nsy7px0frdr2asmcy0lyfbj73ic1f5yf4q65hrgvryx'';
```
Comment (SuperSandro2000) [inline pkgs/development/lisp-modules/quicklisp-to-nix-output/jonathan.nix:12]: Those should be single quotes.

**Comment by SuperSandro2000** [comment]
I don't think it is a good idea to self merge a PR on 4 hour notice that touches 316(!) files.

**Comment by 7c6f434c** [comment]
The number is meaningless, as the files are generated, we just split them unlike the infinite-length generated Node package list files. The large-file-count change is rerunning a generator that has not changed beyond the most minor fixes for a long time.

Also, full rebuild of all lispPackages for SBCL takes less than a medium-sized C++ package, so it's not like it;s a lot of code. And Quicklisp is an ecosystem where application code keeps working across updates with no changes.

I haven't actually changed that part of the generator. Maybe someone did a treewide including generated files. Anyway all the files will be change on any update, Quicklisp version is there in every file.

**Comment by SuperSandro2000** [comment]
> Quicklisp version is there in every file.

Maybe that is something to improve in the future.

**Comment by 7c6f434c** [comment]
Or maybe it is the latest Quicklisp version where the change has happenned. But anyway upstream Quicklisp repository has monthly releases, so the updates are mass-updates even if we have only a subset.

**Pull request merged**